### DISCUSSION TITLE
End of Life - New Relic Terraform providers before version 2.37.0

### DISCUSSION THREAD

**Discussion by kidk** [newrelic/terraform-provider-newrelic#1667] [Announcements]
Hi community 👋 

Just wanted to make everyone aware of this ticket https://github.com/newrelic/terraform-provider-newrelic/issues/1665 which concerns an EOL that will impact all version of the New Relic Terraform provider before version 2.37.0

If you have feedback or concerns, feel free to respond to the ticket or in this thread. 
Thanks a lot
The observability as code team

**Comment by rdhar** [comment]
If possible, would be keen to read more about the reasons leading up to the decision. I suppose it might be to do with the pre-v1 Terraform provider's EOL.

**Reply by kidk** [reply]
Absolutely, we're still preparing this EOL but once we're ready we'll post the full summary and timeline. I wanted to let everyone know as soon as possible.

**Reply by kidk** [reply]
With some delay: The entity types for some New Relic entities will be changing to support Open Telemetry and other future innovations. The problem is that in old versions of Terraform/CLI the entity type is hardcoded because Golang didn't support generics at the time. This means once the entity type is changed on the New Relic platform API (planned for end of Aug), any Terraform resource that interacts with the entity type code will fail, leading us to push everyone to upgrade to a more recent version of our provider. 

Hope this helps. Let me know if you have any questions.

**Reply by rdhar** [reply]
There'll undoubtedly be teething issues in the beginning but, fingers-crossed, it brings the Terraform provider closer to managing a wider breadth of entities across New Relic at scale.
